### ISSUE TITLE
 The change for the regional GSI treatment of  pressure in the fv3-lam interface 

### ISSUE THREAD

**Issue by TingLei-NOAA** [NOAA-EMC/GSI#132]
It is recently clarified in the warm start files of fv3, the pressure ( in terms of delp) is total mass based pressure (including moisture air plus all hydrometeors), while it has been treated as moisture air pressure in the internal GSI process.  
The similar change are also needed for coldstart files (generated from chgres_cube), which will also depend on how it is specified in  the coldstart files (chgres_cube).  The latter is still being discussed. 
This issue will document status for this change.

**Comment by CatherineThomas-NOAA** [comment]
@TingLei-NOAA Yes, the warm starts are total mass pressure, but the model is also expecting total mass pressure increments.  Once the guess pressure is read in, it should be mostly consistent from there.  Did you find any cases other than the load_geop_hgt compressibility that you previously mentioned where pressure is not treated as total mass within GSI?  

There are several points to keep in mind when considering the pressure assimilation for regional:
1. The bigger problem is the global since input and output are not expected to be the same, which is not an issue in the regional
2. The hydrometeor part is very small compared to the rest of the pressure
3. We are already inconsistent since we compute a hydrostatic increment for a nonhydrostatic variable
4. The amount of effort to dig through the GSI is likely moot because the GSI is going away anyway.  We should focus our efforts in making sure that JEDI is consistent.

**Comment by TingLei-NOAA** [comment]
Hi, Cathy,
Thanks for joining this discussion.
Yes, from what I follow through GSI,  except for the q_diag you pointed out
(where pressure = moisture air plus cloud ),
the rest confirm to the moisture air pressure definition, including for
analysis of surface pressure observation.
And, Mingjing also clarify the conversion between moisture air pressure and
total mass pressure is outside of gsi .
So, I think this makes it clear that we should update accordingly for
regional gsi interface for warmstart files.
This will not interfere with what global gsi is doing (since there are no
related conversion in the global GSI runs).
And, even if there are some changes for global interface for fv3 , which
will not interfer with our regional GSI fv3-lam interface
only if all changes are done in the IO step without changes to internal
processes like ges_q  and q control variables.

So, here what regional GSI is planning to do is separate from global gsi
will do (only if we follow the "tradition" -- moisture air pressure is used
in the GSI).
Hope my explanation help some clarification.

Do you think we can have a google meeting to, hopefully, resolve any
existing confusions?
Really appreciate your help through this process.
Regards,
Ting




______________________________

Ting Lei

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Cubicle 2765

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3624

**Comment by TingLei-NOAA** [comment]
And, for regional GSI, more focus is on severe weather, in which the
correct treatment of hydrometeors in association with pressure is expected
to be important.
Ting

______________________________

Ting Lei

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Cubicle 2765

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3624

**Comment by TingLei-NOAA** [comment]
cathy,
 I think I  can first keep this changes in my fork  and see what the impact
would be from our planned comparison experiments,
This will give more evidence and time to resolve this issue.
Thanks.
Ting

______________________________

Ting Lei

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Cubicle 2765

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3624

**Comment by CatherineThomas-NOAA** [comment]
Thanks, Ting.  I appreciate the attention given to the separation of global/regional concerns.

> the rest confirm to the moisture air pressure definition, including for
> analysis of surface pressure observation.

Can you point me to some of the code that shows this?  I'm having trouble finding it myself.

**Comment by TingLei-NOAA** [comment]
Cathy,
In addition to the hgt derived from the pressure., I think one example
is genqsat subroutine, which calculates the saturated q from pressure
/temperature following the normal procedure. If the pressure there include
hydrometeors , some changes are needed.
Also, If pressure includes hydrometeors, we will see quite some changes
needed in, like, control2state_ad.f90, for calculating adjoint for cloud
water and hydrometors control variables.
Basically, to allow for the pressure control variable to include other
control variable(cloud water/hydrometeors)  is a complicating method.
Hope this helps .
Regards,
Ting



______________________________

Ting Lei

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Cubicle 2765

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3624

**Comment by TingLei-NOAA** [comment]
The changes for GSI interface for FV3-LAM (regional GSI)  itself has been added and the impact had been  tested preliminarily (see attached to get an impression of differences from the "new" treatment")(https://github.com/TingLei-NOAA/regional_workflow.git). 
[Emc-report-gsi treatment of total mass pressure.docx](https://github.com/NOAA-EMC/GSI/files/6221820/Emc-report-gsi.treatment.of.total.mass.pressure.docx)
 As expected, the differences of magnitude are relatively small.  Its' systematic impacts are going to be tested in the near future in comparison with fv3-cam parallel run.  
The corresponding changes in GSI EnKF are to be done next, which is expected to be of less impacts than in GSI when, to name a few,  the former is using  observation operators  from GSI

**Comment by TingLei-NOAA** [comment]
A correction:  the current fork containing the changes is https://github.com/TingLei-NOAA/GSI, which has just been merged with current GSI EMC master.

**Comment by TingLei-NOAA** [comment]
The similar changes for cold start files (generated from chgres_cube) is pending further plans decided for those data.
